conveyal/analysis-ui
# Temporary access group rights for admins

This PR add's a page allowing administrators to set a temporary access group to help with debugging customer accounts.

It stores the access group in a session cookie which ties the temporary access group to the current browser session. When the browser is closed it will be reset. This allow's for using this feature in incognito mode so it will not affect your default browser session.

Depends on https://github.com/conveyal/analysis-internal/pull/45

## Test plan 1

1. Open a private session in your browser.
1. Sign in as an administrator. 
1. Go to `/admin/set-access-group`.
1. Set the access group to the test group. Ex: `test-group`
1. Navigate to the home page. Ensure that the temporary test group name is displayed.
1. Create/Read/Update/Delete a region, project, and modification.
1. Set the access group back to the admin group.
1. Navigate to the home page. Should see the admin groups resources.

## Test plan 2

1. Open a private session.
1. Sign in as a non-administrator.
1. Ensure that you are unable to go to `/admin/set-access-group`.
1. Confirm inability to navigate to a region for a separate access group.
